File: from-0201-to-0300/0232-implement-queue-using-stacks/solution.py
class MyQueue:

    def __init__(self):
        self.stack = []
        self.backup = []

    def push(self, x: int) -> None:
        self.stack.append(x)

    def pop(self) -> int:
        while self.stack:
            self.backup.append(self.stack.pop())
            
        popval = self.backup.pop()
        
        while self.backup:
            self.stack.append(self.backup.pop())
            
        return popval

    def peek(self) -> int:
        while self.stack:
            self.backup.append(self.stack.pop())
            
        peekval = self.backup[-1]
        
        while self.backup:
            self.stack.append(self.backup.pop())
            
        return peekval

    def empty(self) -> bool:
        return len(self.stack) == 0


# Your MyQueue object will be instantiated and called as such:
# obj = MyQueue()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.peek()
# param_4 = obj.empty()

File: from-0201-to-0300/0232-implement-queue-using-stacks/solution.go
type MyQueue struct {
    S1 []int
    S2 []int
}


func Constructor() MyQueue {
    return MyQueue{}
}


func (this *MyQueue) Push(x int)  {
    this.S1 = append(this.S1, x)
}


func (this *MyQueue) Pop() int {
    if len(this.S2) == 0 {
        for len(this.S1) != 0 {
            index := len(this.S1) - 1
            element := this.S1[index]
            this.S1 = this.S1[:index]
            this.S2 = append(this.S2, element)
        }
    }
    
    index := len(this.S2) - 1
    element := this.S2[index]
    this.S2 = this.S2[:index]
    
    return element
}


func (this *MyQueue) Peek() int {
    if len(this.S2) == 0 {
        for len(this.S1) != 0 {
            index := len(this.S1) - 1
            element := this.S1[index]
            this.S1 = this.S1[:index]
            this.S2 = append(this.S2, element)
        }
    }
    
    index := len(this.S2) - 1
    element := this.S2[index]
    
    return element
}


func (this *MyQueue) Empty() bool {
    if len(this.S1) == 0 && len(this.S2) == 0 {
        return true
    }
    return false
}


/**
 * Your MyQueue object will be instantiated and called as such:
 * obj := Constructor();
 * obj.Push(x);
 * param_2 := obj.Pop();
 * param_3 := obj.Peek();
 * param_4 := obj.Empty();
 */